From 8f982399b6cdf297407263d4ca365ad8bf7e9cbc Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Thu, 13 Jun 2019 21:39:19 -0700
Subject: [PATCH] Append LTCG arguments properly

If the QMAKE_CXXFLAGS already had -ffat-lto-objects, the *= wouldn't add
it it again, after our "-flto -fno-fat-lto-objects", which meant the
last one would stand.

Change-Id: Ic9cfa6256b5045caa6e6fffd15a7f6cda7aaa837
---
 mkspecs/features/ltcg.prf | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mkspecs/features/ltcg.prf b/mkspecs/features/ltcg.prf
index ccf0226272..f2dbc6fa9e 100644
--- a/mkspecs/features/ltcg.prf
+++ b/mkspecs/features/ltcg.prf
@@ -28,7 +28,10 @@ CONFIG(release, debug|release) {
     }
 
     load(link_ltcg)
-    QMAKE_CFLAGS *= $$QMAKE_CFLAGS_LTCG
-    QMAKE_CXXFLAGS *= $$QMAKE_CXXFLAGS_LTCG
-    QMAKE_LFLAGS *= $$QMAKE_LFLAGS_LTCG
+    QMAKE_CFLAGS -= $$QMAKE_CFLAGS_LTCG
+    QMAKE_CFLAGS += $$QMAKE_CFLAGS_LTCG
+    QMAKE_CXXFLAGS -= $$QMAKE_CXXFLAGS_LTCG
+    QMAKE_CXXFLAGS += $$QMAKE_CXXFLAGS_LTCG
+    QMAKE_LFLAGS -= $$QMAKE_LFLAGS_LTCG
+    QMAKE_LFLAGS += $$QMAKE_LFLAGS_LTCG
 }
-- 
2.22.0

